src/components/hmi_message_handler/src/websocket_session.cc

```diff
@@ -0,0 +1,332 @@
+                  } else
+                  {
+                     DBG_MSG_ERROR(("Not possible to initialize mControllersIdStart!\n"));
+                  }
+               } else if ("MB.subscribeTo" == method || "MB.unregisterComponent" == method || "MB.unsubscribeFrom" == method)
+               {
+                  //nothing to do for now
+               } else
+               {
+                  controller_->processResponse(method, message);
+               }
+            } else {
+               DBG_MSG_ERROR(("Request with id %s has not been found!\n", id.c_str()));
+            }
+         } else {
+            DBG_MSG(("Message is request!\n"));
+            std::string method = message["method"].asString();
+            std::string component_name = GetComponentName(method);
+
+            if(component_name == "MB") {
+               controller_->processInternalRequest(message, this);
+            } else {
+               controller_->pushRequest(message, this);
+               controller_->processRequest(message);
+            }
+         }
+      } else {
+         DBG_MSG_ERROR(("Message contains wrong data!\n"));
+         sendJsonMessage(error);
+      }         
+   }
+
+   bool WebsocketSession::isNotification(Json::Value& root) {
+      DBG_MSG(("CMessageBrokerController::isNotification()\n"));
+      bool ret = false;
+      if (false == root.isMember("id"))
+      {
+         ret = true;
+      }
+      DBG_MSG(("Result: %d\n", ret));
+      return ret;
+   }
+
+   bool WebsocketSession::isResponse(Json::Value& root) {
+      DBG_MSG(("CMessageBrokerController::isResponse()\n"));
+      bool ret = false;
+      if ((true == root.isMember("result")) || (true == root.isMember("error")))
+      {
+         ret = true;
+      }
+      DBG_MSG(("Result: %d\n", ret));
+      return ret;
+   }
+
+   std::string WebsocketSession::findMethodById(std::string id) {
+      DBG_MSG(("CMessageBrokerController::findMethodById()\n"));
+      sync_primitives::AutoLock auto_lock(queue_lock_);
+      std::string res = "";
+      std::map <std::string, std::string>::iterator it;
+      it = mWaitResponseQueue.find(id);
+      if (it != mWaitResponseQueue.end())
+      {
+         res = (*it).second;
+         mWaitResponseQueue.erase(it);
+      }
+      return res;
+   }
+
+   bool WebsocketSession::checkMessage(Json::Value& root, Json::Value& error)
+   {
+      DBG_MSG(("CMessageBrokerController::checkMessage()\n"));
+      Json::Value err;
+
+      try
+      {
+         /* check the JSON-RPC version => 2.0 */
+         if (!root.isObject() || !root.isMember("jsonrpc") || root["jsonrpc"] != "2.0")
+         {
+            error["id"] = Json::Value::null;
+            error["jsonrpc"] = "2.0";
+            err["code"] = NsMessageBroker::INVALID_REQUEST;
+            err["message"] = "Invalid MessageBroker request.";
+            error["error"] = err;
+            return false;
```

Comment:
Seems like this could be filled out by another function, since this block is repeated several paces.